File: customer/src/main/java/com/shieldteq/customer/audit/AuditAwareImpl.java
package com.shieldteq.customer.audit;

import io.micrometer.common.lang.NonNullApi;
import org.springframework.data.domain.AuditorAware;
import org.springframework.stereotype.Component;

import java.util.Optional;

@NonNullApi
@Component("auditAwareImpl")
public class AuditAwareImpl implements AuditorAware<String> {
    public Optional<String> getCurrentAuditor() {
        return Optional.of("ADMIN");
    }
}

File: inventory/src/main/kotlin/com/shieldteq/inventory/audit/AuditAwareImpl.kt
package com.shieldteq.inventory.audit

import io.micrometer.common.lang.NonNullApi
import org.springframework.data.domain.AuditorAware
import org.springframework.stereotype.Component
import java.util.*

@NonNullApi
@Component("auditAwareImpl")
class AuditAwareImpl : AuditorAware<String> {
    override fun getCurrentAuditor(): Optional<String> = Optional.of("ADMIN")
}